crates/ruff_linter/src/rules/mccabe/rules/function_is_too_complex.rs

```diff
@@ -96,8 +97,16 @@ fn get_complexity_number(stmts: &[Stmt]) -> usize {
                 complexity += get_complexity_number(orelse);
             }
             Stmt::Match(ast::StmtMatch { cases, .. }) => {
-                for case in cases {
+                let last_index = cases.len() - 1;
```

Comment:
Can we use `saturating_sub` here to avoid a panic for `match a:`. This is not valid python but the parser might soon be able to recover from incomplete statements.
```suggestion
                let last_index = cases.len().saturating_sub(1);
```

Reply:
Went with Dhurv's suggestion here.